src/Agent.Listener/Configuration/AutoLogonManager.cs

```diff
@@ -75,6 +75,28 @@ public override void Initialize(IHostContext hostContext)
                 _terminal.WriteError(StringUtil.Loc("InvalidAutoLogonCredential"));
             }
 
+            _autoLogonRegManager.GetAutoLogonUserDetails(out string currentAutoLogonUserDomainName, out string currentAutoLogonUserName);
+
+            if (currentAutoLogonUserName != null
+                    && !userName.Equals(currentAutoLogonUserName, StringComparison.CurrentCultureIgnoreCase)
+                    && !domainName.Equals(currentAutoLogonUserDomainName, StringComparison.CurrentCultureIgnoreCase))
+            {
+                string currentAutoLogonAccount = String.Format("{0}\\{1}", currentAutoLogonUserDomainName, currentAutoLogonUserName);
+                if (string.IsNullOrEmpty(currentAutoLogonUserDomainName) || currentAutoLogonUserDomainName.Equals(".", StringComparison.CurrentCultureIgnoreCase))
+                {
+                    currentAutoLogonAccount = String.Format("{0}\\{1}", Environment.MachineName, currentAutoLogonUserName);
+                }
+
+                Trace.Warning($"AutoLogon already enabled for {currentAutoLogonAccount}.");    
+                if(!command.GetOverwriteAutoLogonSettings(currentAutoLogonAccount))
+                {
+                    Trace.Info($"Skipping the autologon configuration.");
+                    _terminal.WriteLine(StringUtil.Loc("SkipAutoLogonConfiguration"));
```

Comment:
do we need error out when user choose to not overwrite? noticed, not overwrite is the default option.